### ISSUE TITLE
Memory leak in sapper export

### ISSUE THREAD

**Issue by nealalpert** [sveltejs/sapper#851]
**Description:**
Sapper export's recursive fetching of URLs can exceed node's write capacity resulting in very large memory footprint.

The [handle function's](https://github.com/sveltejs/sapper/blob/master/src/api/export.ts#L138) recursive page crawling and URL fetch calls can end up overwhelming the [save function's](https://github.com/sveltejs/sapper/blob/master/src/api/export.ts#L105) fs.writeFileSync call's capacity to write to disk resulting in the yootil's queue memory footprint to exceed node's allowed amount.

**Steps To Reproduce:**
Run `sapper export --legacy` with a site with 30,000 pages where some of those pages are driven by sizable amounts of data.  Our current structure looks a bit like:

```
.
└── routes
    └── [lang]
        ├── things
        │   ├── [slug]
        │   │   ├── prices
        │   │   │   ├── _prices.js
        │   │   │   ├── index.json.js
        │   │   │   ├── index.svelte
        │   │   │   └── shop.json.js
        │   │   ├── index.json.js
        │   │   └── index.svelte
        │   ├── _validThings.js
        │   ├── index.json.js
        │   └── index.svelte
        ├── _error.svelte
        ├── _layout.js
        ├── _layout.svelte
        ├── layout.json.js
        └── index.svelte
```

I can't set up a repo with an example at the moment but I'll update this ticket once I can come up with an example that doesn't expose my company's data.

**Expected behavior**
Perhaps throttle the page crawling if the process is at risk of running out of memory?

**Version Info:**
Sapper 0.27.8
Svelte 3.8.0
Rollup 1.19.4

**Severity**
Moderate.  A work around is possible by passing the `--max-old-space-size` optional param to node and upping the memory allotment  when running `sapper export` if the environment can spare it.

**Stacktrace:**
<details>
<summary>JS stack trace</summary>
==== JS stack trace =========================================

Security context: 0x301abcea57c1 <JSObject>
    1: collapseWhitespaceSmart(aka collapseWhitespaceSmart) [/Users/nealalpert/Documents/svelte-poc/node_modules/html-minifier/src/htmlminifier.js:~73] [pc=0x23716a9648d6](this=0x301ac9e822d1 <undefined>,str=0x301abceea769 <String[1]:  >,prevTag=0x301a54aeea99 <String[9]: /noscript>,nextTag=0x301a54aeeb79 <String[5]: /head>,options=0x301ad30edcc1 <Object map = 0x301aa77858a9>)
    2: chars [/Use...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 4: v8::internal::Factory::NewFixedArray(int, v8::internal::PretenureFlag) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 5: v8::internal::HashTable<v8::internal::StringTable, v8::internal::StringTableShape>::EnsureCapacity(v8::internal::Handle<v8::internal::StringTable>, int, v8::internal::PretenureFlag) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 6: v8::internal::StringTable::LookupKey(v8::internal::Isolate*, v8::internal::StringTableKey*) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 7: v8::internal::StringTable::LookupString(v8::internal::Isolate*, v8::internal::Handle<v8::internal::String>) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 8: v8::internal::Runtime_KeyedGetProperty(int, v8::internal::Object**, v8::internal::Isolate*) [/Users/nealalpert/.nvm/versions/node/v8.11.1/bin/node]
 9: 0x23716a3042fd
10: 0x23716a9648d6
</details>

**Comment by khrome83** [comment]
We also validated that `--concurrent` does not modify this due to the bottleneck and did not improve performance. We tested increasing the queue to from the default of 8 to 16, and 32. We also validated setting it to 1 did not resolve anything.

**Comment by Conduitry** [comment]
Fixed in 0.27.14, via #869, finally! Thank you!

**Comment by sw-yx** [comment]
was almost a year's wait with no feedback. would recommend adding someone on @khrome83's team to help maintain and/or having more formal governance for Sapper, to give companies like @khrome83's confidence when making tech stack decisions.
